server/TrillianLogServer.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/storage"
+	"golang.org/x/net/context"
+	"sync"
+)
+
+var (
+	// TODO: Delete when implementation done
+	// ErrNotImplemented is returned when an operation is not supported yet
+	ErrNotImplemented = errors.New("Not yet implemented")
+)
+
+// LogStorageProviderFunc decouples the server from storage implementations
+type LogStorageProviderFunc func(int64) *storage.LogStorage
```

Comment:
Could the impl of this func reasonably fail and need to return an error (unknown ID, DB unavailable, etc.)?